Sources/TuistKit/Utils/Target+PlatformResolution.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+import TuistGraph
+
+struct UnspecifiedPlatformError: Error, CustomStringConvertible {
+    let target: Target
+    var description: String {
+        "Only single platform targets supported. The target \(target.name) specifies multiple supported platforms (\(target.supportedPlatforms.map(\.rawValue).joined(separator: ", ")))."
+    }
+}
+
+extension Target {
+    /// Platform for use with services when a platform is not specified
+    var servicePlatform: Platform {
```

Comment:
I find `servicePlatform` not very representative of what the platform represents but I can't come up with a better name 😬. What about `defaultPlatform`?